### ISSUE TITLE
vela show on local definitions does not work (no cluster case)

### ISSUE THREAD

**Issue by EikeHens** [kubevela/kubevela#5560]
**Describe the bug**
vela show on local definitions does not work (no cluster case)
vela show <my-local-definitions>
Error: Get "https://0.0.0.0:34997/api?timeout=32s": dial tcp 0.0.0.0:34997: connect: connection refused

**To Reproduce**
Remove cluster connections e.g. kubectl
vela show <my-local-definitions>
Error: Get "https://0.0.0.0:34997/api?timeout=32s": dial tcp 0.0.0.0:34997: connect: connection refused

**Expected behavior**
Create documentation from a custom addon stored in the file system in a pipeline

**KubeVela Version**
vela version: CLI Version: v1.6.1

**Comment by EikeHens** [comment]
**Update**
I tested this now also with vela version 1.7.1. Same behavior but slightly different error message:

12:22:44 score@md3hgtyc /mnt/c/development/IoTPlatform/software/score/packages/essentials $ vela show definitions 
Error: invalid configuration: no configuration has been provided, try setting KUBERNETES_MASTER environment variable
12:23:11 score@md3hgtyc /mnt/c/development/IoTPlatform/software/score/packages/essentials $ vela show definitions/components/
Error: invalid configuration: no configuration has been provided, try setting KUBERNETES_MASTER environment variable
12:23:28 score@md3hgtyc /mnt/c/development/IoTPlatform/software/score/packages/essentials $ vela show definitions/components/score-webservice.yaml 
Error: invalid configuration: no configuration has been provided, try setting KUBERNETES_MASTER environment variable
12:23:42 score@md3hgtyc /mnt/c/development/IoTPlatform/software/score/packages/essentials $ vela version
CLI Version: 1.7.3
Core Version:
GitRevision: git-f49f11dd
GolangVersion: go1.19.5

**Comment by wonderflow** [comment]
This should be a bug that we shouldn't rely on cluster connection for `vela show` with local definitions. \cc @chivalryq

**Comment by bhavyastar** [comment]
Hey, @wonderflow I want to work on this issue! Can you assign this to me?

**Comment by Sajiyah-Salat** [comment]
hello @bhavyastar are you working on this? If not I would like to take up this issue.

**Comment by Sajiyah-Salat** [comment]
Great. No worries.

**Comment by EikeHens** [comment]
Is there any update? We need to generate a offline documentation. Or is there any workaround?

**Comment by chivalryq** [comment]
@EikeHens As a workaround you can run a kind cluster when generating. BTW @bhavyastar do you have any progress on this?

**Comment by EikeHens** [comment]
@bhavyastar Do you have any progress on this issue? We need to document our addons somehow in a offline way. Online generation works, but we need to have it offline in our pipelines. Can I do something to speed up the process?

**Comment by wonderflow** [comment]
@EikeHens Yes, please go ahead. It's two weeks since last ping. I'll remove the assignment.

Thanks to @bhavyastar , you can also find other good first issues .

**Comment by EikeHens** [comment]
@wonderflow Not sure if I can take over the bug fix.

**Comment by briankane** [comment]
This issue has been inactive for over 2 years and is scheduled for closure as part of our backlog maintenance.

  **If this issue is still relevant:**
  - Please comment below to keep it open
  - Provide updated context if the original description is outdated

  **Timeline:** If there's no activity by **Friday, October 24th**, this issue will be automatically closed.

  **Note:** Issues can always be reopened if they become relevant again. We encourage the community to reopen and reprioritize any incorrectly 
  closed issues.

**Comment by briankane** [comment]
This issue has been closed following the stale issue notice. 

If this issue is still relevant and should remain open, please:
1. Reopen the issue
2. Remove the 'stale' label
3. Provide an update on the current status or importance

Thank you for your understanding and contribution to KubeVela!
